v2/pkg/genruntime/configmaps/validation.go

```diff
@@ -39,6 +45,32 @@ func ValidateDestinations(destinations []*genruntime.ConfigMapDestination) (admi
 		locations.Add(pair)
 	}
 
+	if destinationExpressions == nil {
+		return nil, nil
+	}
+
+	for _, dest := range destinationExpressions {
+		if dest == nil {
+			continue
+		}
+
+		if dest.Key == "" {
+			// TODO: Key may be empty because of map[string]string supported exports.
+			// TODO: We should validate that in more depth but need a CEL parser to do so.
+			continue
+		}
+
+		pair := keyPair{
```

Comment:
Consider adding a `createKeyPair()` method to `core.DestinationExpression()` (and to `genruntime.ConfigMapDestination`) to simplify these loops.

Reply:
Given the simplicity of the structure and the fact that it's not exported, I opted to keep the pure struct-initialization pattern here rather than do a `new` or `make` method as it felt like it protected better against the main bug you could have here of getting `key` and `value` ordering mixed up.